src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -461,10 +461,23 @@ private void parseDoc(Map<String, String> namespacePrefixesByUri) {
                 final String instanceId = instanceNodeIdStrs.get(instanceIndex);
                 final String instanceSrc = instance.getAttributeValue(null, "src");
 
-                if (instanceSrc != null && instanceSrc.toLowerCase().startsWith("jr://file")) { // file or file-csv
+                final String externalSrc;
+                if (instanceSrc == null) {
+                    // It's internal, not external.
+                    externalSrc = null;
+                } else if (instanceSrc.toLowerCase().startsWith("jr://file")) { // file or file-csv
```

Comment:
Strictly speaking, shouldn't this check for `startsWith("jr://file/")`?

Reply:
I believe this is to support both the last-saved instance from this PR, and the external csv itemsets feature, which uses a `file-csv` "host". 

Now that you mention it, maybe we should separate both cases to improve semantics at the expense of some duplication? Is this what you had in mind, @zestyping?

Reply:
That seems like a good idea.  Certainly, reading the code as it is now, it's not evident that it's intended to accept `jr://file/abcd` and `jr://file-csv/abcd`.